api/src/main/java/org/apache/iceberg/expressions/ManifestEvaluator.java

```diff
@@ -245,12 +245,12 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
     }
 
     @Override
-    public <T> Boolean in(BoundReference<T> ref, Literal<T> lit) {
+    public <T> Boolean in(BoundReference<T> ref, LiteralSet<T> literalSet) {
       return ROWS_MIGHT_MATCH;
```

Comment:
Since we can translate an `in` predicate to equality predicates or'ed together, this should call `eq` and or the results together.

Reply:
@rdblue 
It is probably better to be implemented in the next PR, which I am going to work on it after this one.
Please let me know your thoughts. thanks.